plugin/storage/memory/memory.go

```diff
@@ -107,20 +107,16 @@ func (st *Store) GetDependencies(ctx context.Context, endTs time.Time, lookback
 		if traceIsBetweenStartAndEnd(startTs, endTs, trace) {
 			for _, s := range trace.Spans {
 				parentSpan := findSpan(trace, s.ParentSpanID())
-				if parentSpan != nil {
-					if parentSpan.Process.ServiceName == s.Process.ServiceName {
-						continue
-					}
-					depKey := parentSpan.Process.ServiceName + "&&&" + s.Process.ServiceName
-					if _, ok := deps[depKey]; !ok {
-						deps[depKey] = &model.DependencyLink{
-							Parent:    parentSpan.Process.ServiceName,
-							Child:     s.Process.ServiceName,
-							CallCount: 1,
-						}
-					} else {
-						deps[depKey].CallCount++
-					}
+				if parentSpan != nil && parentSpan.Process.ServiceName != s.Process.ServiceName {
+					updateServiceDependencyLinks(parentSpan.Process.ServiceName, s.Process.ServiceName, deps)
+				}
+				// Handle client spans separately
+				if isClientSpan(s) {
+					childService := inferServiceName(s, false)
```

Comment:
I don't think this is needed. All you're adding is a use case where both `isClientSpan && isLeaf` are true.

Reply:
I've removed the redundant check for cases where a span is both a client and a leaf, simplifying the logic.